src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarEntry.cs

```diff
@@ -578,6 +587,17 @@ private async Task ExtractAsRegularFileAsync(string destinationFileName, Cancell
                 PreallocationSize = Length,
                 Options = FileOptions.Asynchronous
             };
+            if (OperatingSystem.IsWindows())
```

Comment:
Can we extract creating the FileStreamOptions into a separate method and share it with the sync version above? It looks like the only difference is the `FileOptions.Asynchronous` piece.

And it looks like this copy of the code has a bug because we are setting PreallocationSize on line 587, and then setting it again here.